datafusion/src/execution/runtime_env.rs

```diff
@@ -25,47 +25,110 @@ use crate::{
+    pub object_store_registry: Arc<ObjectStoreRegistry>,
+    /// DataFusion task contexts that are registered within the Executors' Runtime
+    pub task_context_registry: Option<Arc<TaskContextRegistry>>,
+    /// DataFusion session contexts that are registered within the Scheduler's Runtime
+    pub session_context_registry: Option<Arc<SessionContextRegistry>>,
 }
 
-impl Debug for RuntimeEnv {
-    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
-        write!(f, "RuntimeEnv")
+impl RuntimeEnv {
+    /// Create an executor env based on configuration
+    pub fn new_executor_env(config: RuntimeConfig, executor_id: String) -> Result<Self> {
+        let RuntimeConfig {
+            memory_manager,
+            disk_manager,
+        } = config;
+        Ok(Self {
+            executor_id: Some(executor_id),
+            is_local: false,
+            memory_manager: MemoryManager::new(memory_manager),
+            disk_manager: DiskManager::try_new(disk_manager)?,
+            object_store_registry: Arc::new(ObjectStoreRegistry::new()),
+            task_context_registry: Some(Arc::new(TaskContextRegistry::new())),
+            session_context_registry: None,
+        })
     }
-}
 
-impl RuntimeEnv {
-    /// Create env based on configuration
-    pub fn new(config: RuntimeConfig) -> Result<Self> {
+    /// Create a scheduler env based on configuration
+    pub fn new_scheduler_env(config: RuntimeConfig) -> Result<Self> {
         let RuntimeConfig {
-            batch_size,
             memory_manager,
             disk_manager,
         } = config;
+        Ok(Self {
+            executor_id: None,
+            is_local: false,
+            memory_manager: MemoryManager::new(memory_manager),
+            disk_manager: DiskManager::try_new(disk_manager)?,
+            object_store_registry: Arc::new(ObjectStoreRegistry::new()),
+            task_context_registry: None,
+            session_context_registry: Some(Arc::new(SessionContextRegistry::new())),
+        })
+    }
 
+    /// Create a local env based on configuration
+    pub fn new_local_env(config: RuntimeConfig) -> Result<Self> {
+        let RuntimeConfig {
+            memory_manager,
+            disk_manager,
+        } = config;
         Ok(Self {
-            batch_size,
+            executor_id: None,
+            is_local: true,
             memory_manager: MemoryManager::new(memory_manager),
             disk_manager: DiskManager::try_new(disk_manager)?,
+            object_store_registry: Arc::new(ObjectStoreRegistry::new()),
+            task_context_registry: None,
+            session_context_registry: Some(Arc::new(SessionContextRegistry::new())),
         })
     }
 
-    /// Get execution batch size based on config
-    pub fn batch_size(&self) -> usize {
-        self.batch_size
+    /// Return the global singleton RuntimeEnv
+    pub fn global() -> &'static Arc<RuntimeEnv> {
```

Comment:
I am concerned that if we have this function, it will become too easy for DataFusion code to accidentally use the global `RuntimeEnv` rather than the one that is passed through

I would prefer that the global runtime is put in ballista (or a separate module in DataFusion)